service_capacity_modeling/interface.py

```diff
@@ -756,6 +762,122 @@ class CurrentClusters(ExcludeUnsetModel):
+    # "Dataset" related buffers, e.g. Disk and Memory
+    storage = "storage"
+    # Specific resources
+    cpu = "cpu"
+    network = "network"
+    disk = "disk"
+    # Memory can be used for various buffers
+    memory = "memory"
+    memory_traffic = "memory-traffic"
+    memory_write_cache = "memory-write-cache"
+    memory_read_cache = "memory-read-cache"
+
+
+class Buffer(ExcludeUnsetModel):
+    # The type of buffer, intentionally not BufferComponent to allow
+    # models to define their own.
+    component: str
+    # The value of the buffer expressed as a ratio over "normal" load e.g. 1.5x
+    ratio: float
+    # Set only if the source of the buffer is not the direct component
+    source: Optional[str] = None
+
+
+class Buffers(ExcludeUnsetModel):
+    """Typical buffers (headroom) over the requirements to build into the system
+
+    Note that typically models make these choices, for example if a workload
+    requires 10 CPU cores, but the operator likes to build in 2x buffer for
+    variance (20 cores provisioned), they would express that as:
+        Buffer(
+            desired={
+                "compute": 2.0
+            }
+        )
+
+    There is a series of fallbacks here over common desired buffers:
+        compute -> default
+        cpu -> compute -> default
+        network -> compute -> default
+        disk -> storage -> default
+        memory -> storage -> default
+        memory-traffic -> memory -> default
+        memory-write-cache -> memory -> default
+        memory-read-cache -> memory -> default
+    """
+
+    # The default buffer if a specific buffer isn't known
+    # Models should prefer to document their precise buffers in desired
+    default: float = 1.5
+    # Desired compute, storage, cpu, memory, etc... buffers
+    desired: Dict[str, float] = {}
+    # Derive these buffers from current clusters or model context
+    derived: List[str] = []
+
+    def buffer_for_component(
+        self, component: str, current: Optional[CurrentClusterCapacity] = None
+    ) -> Buffer:
+        # flake8: noqa: C901
+        """Calculates buffer for a given component, handling fallbacks
+
+        Typical usage would be buffer("compute") to get the compute buffer
+        or buffer("storage") to get the storage buffer, but you can ask specific
+        resources like buffer("cpu").
+
+        Returns a tuple of the buffer expressed as a ratio (e.g. 2.0 for 2x buffer)
+        and the source of that buffer, e.g. "compute" if "cpu" was requested but
+        there is not explicit cpu buffer set by the model
+        """
+        buffers = dict(self.desired)
+        if isinstance(component, BufferComponent):
+            component = component.value
+
+        if current is not None:
+            # TODO (jolynch | 2025-02-15) Use current cluster when provided
+            #  If derived is set for the component, we should take buffers
+            #  from the current utilization not from desired. If a cluster is
```

Comment:
We might have to pass derived to get the current utilization and then calculate the buffer for the desired. Above statement will be applicable if we can still pass parameters like. 
```"current_clusters" {"zonal": {"cpu_utilization": {low=24, mid=30, high=35}, ... disk_utilization_gib: ...}```

Reply:
With [this](https://github.com/Netflix-Skunkworks/service-capacity-modeling/pull/113/files#diff-92859329f200266e5ba8f8768e693a29d18d65cc2b71e764b5e3fdbdfd0695faR898) comment, my understanding is current utilization can still be passed. 

In that case why can't we pass desired buffer as null or 0 to stick to the current buffer? Why do we need "derived" parameter here?